### PR TITLE
add information architecture library

### PR DESCRIPTION
**Pull request by inwardmovement**
One element is missing: I can't manage to create a proper half disk, any tips on how to create one?

image

Also for this library it would have been useful to be able to somehow lock an element's size, so that it doesn't scale when the group it's in is resized (for the crossbar element in the "arrow with crossbar" group).

image

### CODE REVIEW

**Comment by dwelle** [comment]
How about?

image

https://excalidraw.com/#json=4807326217273344,uUTeuyc0cdegeLhun-7oKQ

> Also for this library it would have been useful to be able to somehow lock an element's size, so that it doesn't scale when the group it's in is resized (for the crossbar element in the "arrow with crossbar" group).

You can create a feature request in the main repo, but this feels like out of scope for Excalidraw right now, but who knows.

**Comment by inwardmovement** [comment]
Thanks, it's quite more rough than other shapes but I guess we can't do better for now.

image

**Comment by dwelle** [comment]
Your other shapes have "architect" sloppiness, unlike the half-circle, so that's one reason:

image

image

**Comment by lipis** [comment]
I think we can skip the arrows.. WDYT?

**Comment by inwardmovement** [comment]
Indeed! my bad. I also reworked a bit the shape, I think it's ok now:

image

Yes I wondered if we should keep the arrows or not. Maybe it's good to have an "all-in" library so we have all the information architecture shapes available at a glance at the same place. Or only keep the "crossbar arrows" as they are not accessible in one click otherwise?
I don't have a strong opinion about this, just tell me what you think is best and I'll do it.

**Comment by lipis** [comment]
I still vote to remove the single line and all the arrows as we have them out of the box.. and you will always going to need to update their size anyway.. drawing a two point arrow is faster.

**Review by lipis — APPROVED** [review]

**Comment by pclainchard** [comment]
> Thanks, it's quite more rough than the other shapes but I guess we can't do better for now.
> 
> image

Have you tried making a full circle, and then hide one half with white rectangular shape ?
Here a sample with every shape selected :

image

**Comment by lipis** [comment]
> Have you tried making a full circle, and then hide one half with white rectangular shape ?
> Here a sample with every shape selected

Not sure if that's a good idea for the shape as it will have an extra padding.

**Comment by inwardmovement** [comment]
I removed the lines and arrows.

> Not sure if that's a good idea for the shape as it will have an extra padding.

I agree.

I just noticed for some shapes we can't "auto-snap" the arrows to them (or "auto-connect", I don't know how you call this).
But nevermind I think it's just the way Excalidraw works, it's better than nothing like that.

File: libraries.json [hunk 1]
```diff
@@ -124,5 +124,14 @@
     "source": "h7y/dropdowns.excalidrawlib",
     "preview": "h7y/dropdowns.png",
     "date": "2021-01-04"
+  },
+  {
+    "name": "Information Architecture",
+    "description": "A visual vocabulary for describing information architecture and interaction design, by Jesse James Garrett (http://www.jjg.net/ia/visvocab).",
```
Comment (lipis) [inline libraries.json:130]:
```suggestion
    "description": "A visual vocabulary for describing information architecture and interaction design, by Jesse James Garrett.",
```

Comment (lipis) [reply libraries.json:130]: A quick google search leads us to that link.. let's remove is as we don't render links yet..

**Comment by dwelle** [comment]
> I just noticed for some shapes we can't "auto-snap" the arrows to them (or "auto-connect", I don't know what you call that).
> But nevermind I think it's just the way Excalidraw works, it's better than nothing like that.

@ls6 had a smart idea to group those shapes with an invisible (`transparent` fill & stroke) rectangle/ellipse of similar shape.

---

@lipis, should wait with merging this until we implement v2 version of the library https://github.com/excalidraw/excalidraw/issues/2653#issuecomment-757915618?

**Comment by inwardmovement** [comment]
Thanks for the tip

**Comment by lipis** [comment]
@dwelle I think we should merge those two.. it's unknown when we'll release the v2.. it might take a while :) But we will ask the owners if they could update their libraries when it's published.

**Pull request merged**